### PR TITLE
BlackoilWellModel: template Scalar type

### PR DESCRIPTION
**Pull request by akva2**
Extracted from https://github.com/OPM/opm-simulators/pull/5286

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this please

File: opm/simulators/wells/BlackoilWellModelRestart.cpp [hunk 1]
```diff
@@ -260,4 +267,6 @@ loadRestartData(const data::Wells&                 rst_wells,
     }
 }
 
+template class BlackoilWellModelRestart<double>;
```
Comment (atgeirr) [inline opm/simulators/wells/BlackoilWellModelRestart.cpp:270]: Is there a way to find all the explicit instantiations like this? For when we need to add a float instantiation, for example.

Comment (akva2) [reply opm/simulators/wells/BlackoilWellModelRestart.cpp:270]: there will be a define FLOW_INSTANTIATE_FLOAT that you can grep for.

File: opm/simulators/wells/BlackoilWellModelGeneric.cpp [hunk 2]
```diff
@@ -407,7 +414,7 @@ checkGEconLimits(
         return;
     }
 
-    GroupEconomicLimitsChecker<double> checker {
+    GroupEconomicLimitsChecker checker {
```
[the remark this answers is not in the document]
Comment (akva2) [reply opm/simulators/wells/BlackoilWellModelGeneric.cpp:417]: Okay I like CTAD but will revert to explicit then.

**Review by atgeirr — APPROVED** [review]
Looks good and can be merged as is (despite my minor comment about template parameters above).

**Comment by akva2** [comment]
jenkins build this please

**Pull request merged**